### PR TITLE
packaging: qemu-snp-experimental: support host kernel with gmem

### PR DESCRIPTION
**Pull request by niteeshkd**
This is required to allow creation of SNP coco on host kernel (e.g. https://github.com/AMDESE/linux ,branch:snp-host-latest) supporting guest private memory for SNP using gmem.

### CODE REVIEW

**Comment by niteeshkd** [comment]
@ryansavino @fitzthum @fidencio

**Comment by fidencio** [comment]
@niteeshkd, does it require any change in the QEMU command line used to create the VM?

**Comment by niteeshkd** [comment]
> does it require any change in the QEMU command line used to create the VM?

It does not require change in the kata shim to create the SNP coco. If we need to handle the attestation, we will have to modify the shim which we will handle later.

**Comment by fidencio** [comment]
I'll leave the review for those more familiar with those changes, and if an ack is needed from my side, I'll add after they do the review.
Thanks @niteeshkd!

**Review by fitzthum — APPROVED** [review]
Looks good. We'll want to make sure to run to upgrade the CI nodes.

Btw @niteeshkd will this new QEMU still function on the old non-UPM SNP Kernel? If not we should probably also add a note to the coco docs.

Also @danmihai1 might be interested.

**Comment by niteeshkd** [comment]
> Btw @niteeshkd will this new QEMU still function on the old non-UPM SNP Kernel? If not we should probably also add a note to the coco docs.

No, this qemu on non-UPM kernel does not work.  Here is the error it reports.
```
$ sudo journalctl -xeu containerd
...
 Feb 13 18:41:04 amdmilan1 kata[4163390]: time="2024-02-13T18:41:04.079527912Z" level=error msg="qemu-system-x86_64: vm-type snp not supported by KVM"
```

**Comment by fitzthum** [comment]
> No, this qemu on non-UPM kernel does not work. Here is the error it reports.

Ok, we'll just have to be clear that we have a change of host requirements for the next release.

**Comment by fitzthum** [comment]
Also @zvonkok might be interested in this?

**Comment by fidencio** [comment]
> No, this qemu on non-UPM kernel does not work. Here is the error it reports.

Please, add this to the commit message. :-)

**Comment by niteeshkd** [comment]
Fixed building of qemu-snp-experimental using tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh .

File: tools/packaging/static-build/qemu/Dockerfile [hunk 1]
```diff
@@ -60,7 +60,8 @@ RUN apt-get update && apt-get upgrade -y && \
 	    patch \
 	    python \
 	    python-dev \
-	    rsync \
+	    python3-venv\
+	    rsync\
```
Comment (ryansavino) [inline tools/packaging/static-build/qemu/Dockerfile:64]: Can we re-add the space here?

**Review by ryansavino — CHANGES_REQUESTED** [review]

**Review by ryansavino — APPROVED** [review]

**Review by fidencio — APPROVED** [review]
lgtm, thanks @niteeshkd!

**Pull request merged**